huaweicloud/services/aad/resource_huaweicloud_aad_forward_rule.go

```diff
@@ -0,0 +1,256 @@
+	port int) (*rules.Rule, error) {
+	resp, err := rules.List(client, instanceId, advancedIp)
+	if err != nil {
+		return nil, fmt.Errorf("error getting Advanced Anti-DDoS advanced rules: %s", err)
+	}
+
+	if len(resp) == 0 {
+		return nil, golangsdk.ErrDefault404{}
+	}
+	if result := filterForwardRuleFromList(resp, protocol, port); result != nil {
+		return result, nil
+	}
+	return nil, golangsdk.ErrDefault404{}
+}
+
+func resourceForwardRuleRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	client, err := conf.AadV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating Advanced Anti-DDoS V1 client: %s", err)
+	}
+
+	instanceId := d.Get("instance_id").(string)
+	advancedIp := d.Get("ip").(string)
+	protocol := d.Get("forward_protocol").(string)
+	forwardPort := d.Get("forward_port").(int)
+	resp, err := GetForwardRuleFromServer(client, instanceId, advancedIp, protocol, forwardPort)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving Advanced Anti-DDoS forward rule")
+	}
+	tflog.Debug(ctx, "Retrieved Advanced Anti-DDos forward rule: %#v", resp)
+
+	mErr := multierror.Append(nil,
+		d.Set("region", region),
+		d.Set("forward_protocol", resp.ForwardProtocol),
+		d.Set("forward_port", resp.ForwardPort),
+		d.Set("source_port", resp.SourcePort),
+		d.Set("source_ip", resp.SourceIp),
+		d.Set("status", resp.Status),
+		d.Set("lb_method", resp.LbMethod),
+	)
+
+	if mErr.ErrorOrNil() != nil {
+		return diag.Errorf("error setting resource: %s", mErr)
+	}
+	return nil
+}
+
+func resourceForwardRuleUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	region := conf.GetRegion(d)
+	client, err := conf.AadV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating Advanced Anti-DDoS V1 client: %s", err)
+	}
+
+	instanceId := d.Get("instance_id").(string)
+	advancedIp := d.Get("ip").(string)
+
+	protocol := d.Get("forward_protocol").(string)
+	if d.HasChange("forward_protocol") {
+		oldVal, _ := d.GetChange("forward_protocol")
+		protocol = oldVal.(string)
+	}
+	forwardPort := d.Get("forward_port").(int)
+	if d.HasChange("forward_port") {
+		oldVal, _ := d.GetChange("forward_port")
+		forwardPort = oldVal.(int)
+	}
+	resp, err := GetForwardRuleFromServer(client, instanceId, advancedIp, protocol, forwardPort)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "error retrieving Advanced Anti-DDoS forward rule")
+	}
+	tflog.Debug(ctx, "Retrieved Advanced Anti-DDos forward rule: %#v", resp)
+
+	opts := rules.UpdateOpts{
+		ForwardProtocol: d.Get("forward_protocol").(string),
+		ForwardPort:     d.Get("forward_port").(int),
+		SourcePort:      d.Get("source_port").(int),
+		SourceIp:        d.Get("source_ip").(string),
+	}
+	err = rules.Update(client, instanceId, advancedIp, resp.ID, opts)
```

Comment:
how about save  `resp.ID` in the state file, so you don't need to get it here

Reply:
I don't want to expose a useless param.